lib/opentelemetry_ecto.ex

```diff
@@ -0,0 +1,91 @@
+defmodule OpentelemetryEcto do
+  @moduledoc """
+  Telemetry handler for creating OpenTelemetry Spans from Ecto query events.
+  """
+
+  require OpenTelemetry.Tracer
+
+  @doc """
+  Attaches the OpencensusEcto handler to your repo events. This should be called
+  from your application behaviour on startup.
+
+  Example:
+
+      OpentelemetryEcto.setup([:blog, :repo, :query])
+
+  You may also supply the following options in the second argument:
+
+    * `:time_unit` - a time unit used to convert the values of query phase
+      timings, defaults to `:microsecond`. See `System.convert_time_unit/3`
+
+    * `:span_prefix` - the first part of the span name, as a `String.t`,
+      defaults to the concatenation of the event name with periods, e.g.
+      `"blog.repo.query"`. This will always be followed with a colon and the
+      source (the table name for SQL adapters).
+  """
+  def setup(event_name, config \\ []) do
+    :telemetry.attach({__MODULE__, event_name}, event_name, &__MODULE__.handle_event/4, config)
```

Comment:
I think it should have always attach to `:query`, so it would be best to get prefix as an argument and then attach to the required event.

Reply:
Do you mean instead of `OpentelemetryEcto.setup([:blog, :repo, :query])` do `OpentelemetryEcto.setup([:blog, :repo])`?

Reply:
Made sense, so if that is what you meant then this is resolved now.